torch/csrc/jit/custom_operator.h

```diff
@@ -9,8 +9,28 @@
 #include <c10/util/Metaprogramming.h>
 #include <c10/util/TypeList.h>
 
-namespace torch { namespace jit {
+namespace torch {
+namespace jit {
 namespace detail {
+/// Checks the static C++ type `T` for correctness to catch common error cases.
+template <typename T>
+void checkStaticArgumentTypes() {
+  // Give nice error messages for some of the common error cases.
+  // Use a LOUD ERROR MESSAGE SO USERS SEE THE STATIC_ASSERT
+  static_assert(
+      !std::is_same<T, int32_t>::value,
```

Comment:
This is not very robust because it’s not even guaranteed that int and int32_t are the same types in any compiler. Also, it doesn’t guard against other types like long, short, etc. It would be better to check that if the type is integral (there are type traits for this) then it’s an int64_t.

Reply:
It's more about catching the very common case of people writing `int` while we only support `int64_t`, but you're right that it doesn't hurt to guard against the other integral types. I'll update